### ISSUE TITLE
no-console doesn't complain about console functions when calling them indirectly

### ISSUE THREAD

**Issue by Goodwine** [palantir/tslint#3207]
### Bug Report

- __TSLint version__: 5.6.0
- __TypeScript version__: 2.4.2
- __Running TSLint via__: Atom

#### TypeScript code being linted

```ts
let x = [1, 2, 3];
console.log(x);
x.forEach(console.log);
```

with `tslint.json` configuration:

```json
{
  "rules": {
    "no-console": true
  }
}
```

#### Actual behavior
The linter complains `console.log(x);`

#### Expected behavior
The linter complains `console.log(x);` and `x.forEach(console.log);`
The linter should not only listen for `console.*` functions, it should also complain about any property.
